Services/Sources/GitHubSource.cs

```diff
@@ -20,11 +20,47 @@
 
 namespace Services.Sources
 {
+
+    public interface IGitHubSource : ISource
+    {
+
+        /// <summary>
+        /// Gets the source.
+        /// </summary>
+        /// <param name="uri">The URI of the source project.</param>
+        /// <exception cref="NotImplementedException"></exception>
+        void GetSource(Uri uri);
```

Comment:
Maybe use virtual or new? You are overriding the method in ISource without override

Reply:
Don't forget to apply this to all the other methods where you are doing this.